File: episodes/002-clean-code-naming/tests/OverdueTest.java
// Sanity test for episode 002 (Java) — plain assertions, no JUnit needed.
import java.time.LocalDate;

public class OverdueTest {
  public static void main(String[] args) {
    long late = Overdue.daysOverdue(LocalDate.of(2026, 7, 1), LocalDate.of(2026, 7, 12));
    if (late != 11) {
      System.err.println("FAIL: expected 11, got " + late);
      System.exit(1);
    }
    long onTime = Overdue.daysOverdue(LocalDate.of(2026, 7, 12), LocalDate.of(2026, 7, 1));
    if (onTime != 0) {
      System.err.println("FAIL: expected 0, got " + onTime);
      System.exit(1);
    }
    System.out.println("ok - episode 002 java sanity test passed");
  }
}

File: episodes/002-clean-code-naming/tests/overdue.test.js
// Sanity test for episode 002 (JavaScript) — plain Node assert, no framework.
const assert = require("assert");
const { daysOverdue } = require("../src/javascript/overdue");

assert.strictEqual(
  daysOverdue(new Date("2026-07-01"), new Date("2026-07-12")),
  11,
);
assert.strictEqual(
  daysOverdue(new Date("2026-07-12"), new Date("2026-07-01")),
  0,
);

console.log("ok - episode 002 javascript sanity test passed");
